front/src/stores/globalPreferenceStore.ts

```diff
@@ -26,3 +31,30 @@ export const selectedBaseIdAtom = atom((get) => {
   // This case should not happen except on initial render
   return "0";
 });
+
+/** Persisted atom to cache the value for the Match Product form input in the Box Reconciliation View. */
+export const reconciliationMatchProductAtom = atomWithStorage<ReconciliationMatchProductFields>(
```

Comment:
move please in a separate store file since it has nothing to do with gloable preferences